Dotty allows an import in a block to shadow a prior definition

```scala
object A {
   def x = 3

   def y = {
     import B._
     x
   }
}
object B {
  def x = 3
}
```
This fails in scalac with:
```
A.scala:6: error: reference to x is ambiguous;
it is both defined in object A and imported subsequently by
import B._
     x
     ^
one error found
```
but succeeds in dotty (`B.x` is picked).

--- a/src/dotty/tools/dotc/typer/Typer.scala
+++ b/src/dotty/tools/dotc/typer/Typer.scala
@@ -232,7 +232,8 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
           }
         }
         val curImport = ctx.importInfo
-        if (curImport != null && curImport.isRootImport && previous.exists) return previous
+        if (ctx.owner.is(Package) && curImport != null && curImport.isRootImport && previous.exists)
+          return previous // no more conflicts possible in this case
         // would import of kind `prec` be not shadowed by a nested higher-precedence definition?
         def isPossibleImport(prec: Int) =
           prevPrec < prec || prevPrec == prec && (prevCtx.scope eq ctx.scope)

--- a/tests/neg/i1145.scala
+++ b/tests/neg/i1145.scala
@@ -1,0 +1,11 @@
+object A {
+   def x = 3
+
+   def y = {
+     import B._
+     x  // error: ambiguous
+   }
+}
+object B {
+  def x = 3
+}

